### ISSUE TITLE
manpage: use asciidoc instead of ronn

### ISSUE THREAD

**Issue by jpleau** [vivien/i3blocks#186]
Hi.

Debian (and probably Ubuntu as well), will not ship ruby-ronn next release (see: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=796497). The outcome is that i3blocks will not be buildable on those distributions when they release.

It was proposed in this bug report that the asciidoc format should be used instead of markdown / ronn to generate man pages.

Do you think this is something that we can adjust in i3blocks? I don't mind taking the time to rewrite the .ronn file to asciidoc (learning it along the way !), there are a few tools that can convert to manpages: asciidoc, asciidoctor, and probably more

**Comment by quinnuendo** [comment]
You might look at other software that can translate markdown to man pages. 

Pandoc (http://pandoc.org/index.html) is a nice tool. It is actively developed and included in package managers of many distributions. I did a quick test on the i3blocks.1.ronn page and it made a similar output to ronn. 

Of course there will be some minor changes to be done - pandoc expects the title to be in a metadata block with % at the start, and the headers should be starting from level one, and there are a few more tweaks for the output to be the same.

But maybe it's less work this way.

There is an example with the programs man page here: http://pandoc.org/demos.html

**Comment by jpleau** [comment]
I'll definitely keep this in mind. Obviously editing the markdown a little bit is less work than converting to asciidoc

**Comment by jpleau** [comment]
@quinnuendo I went with pandoc, here's an example of what I did: 

https://gist.github.com/jpleau/6a675fa44771800d6bf92b9ab1d12f59

Does it look good to you? I'll submit a PR if yes!

**Comment by kb100** [comment]
@jpleau that file looks good to me, what command do you use to build the man page?

**Comment by jpleau** [comment]
Sorry I should have mentionned it :

Save my gist as i3blocks.1.md, then:

`pandoc -s -t man i3blocks.1.md -o i3blocks.1`

**Comment by quinnuendo** [comment]
Seems OK to me, though there are some minodr differences from what I see in the HEAD of i3blocks man page, such as the options being listed at the top of the page.

But overall I think this would work.

Here are a few changes in formating that I made:
https://gist.github.com/quinnuendo/a0ccb28fd2f41f5b10fa0d3d14f6c704

The lists were not converting properly on my tests, and the urls don't benefit from being in brackets (at least for my test).

I would recommend using the long options when you put it in the Makefile, for better future readability, like `pandoc --standalone --output i3blocks.1 --to man i3blocks.1.md`

Also for those who want to test these locally, you can use `man -l i3blocks.1` to point it to a local file for quick comparison.

**Comment by somospocos** [comment]
I'm not going to lobby for exclusion of pandoc, but I feel obliged to provide an end user perspective.

TLDR: enforcing pandoc as dependency just for converting a single file text manpage feels like a giant overkill. I've read the posts in this issue and I understood the reason for switching from ruby-ronn (which seems like just another overkill to me).

pandoc uses haskell. Imagine a system that has nothing with build or runtime haskell dependencies. For example think Slackware Linux (which doesn't even have built-in package dependency tracking, but don't you dare to bring up any of those jokes about it, I grew a thick skin over years). One wants to install i3blocks. He goes here http://slackbuilds.org/repository/14.2/desktop/i3blocks/ , sees pandoc as dependency. He goes here http://slackbuilds.org/repository/14.2/office/pandoc/ , sees that giant dependency list and goes insane. Building all those things one by one, manually, should take like half of a day. Thankfully, there's a tool called sbopkg which comes with a queue generator helper, so it can build the whole list of dependencies for i3blocks and sbopkg will just handle everything (download the sources, build the Slackware packages and install them). Yet, it will take **A LOT** of time. Compared to the build time of i3blocks itself.

If you open the manpage file generated by pandoc from markup, you can see it doesn't look that scary. So my question is what was the reason for not writing the manpage in its own format (Wikipedia tells me it's called troff/groff) from the start, provided that this would remove pandoc and the mammoth haskell ecosystem as dependency?

**Comment by quinnuendo** [comment]
@somospocos, I understand where you are coming from, with the dependency problems.

The simplest, though maybe not the nicest solution is to just not build the man pages for i3blocks? If I am not mistaken, the Makefile has this as a separate target. The man page can then be obtained from other sources, looking it up on-line, already generated, or just viewing the contents.

A compromise that could potentially be made (depending how these things are looked upon in this repo) is to have the generated man tracked by git, but still keeping pandoc as well, and then making sure that the man is regenerated when needed.

**Comment by kb100** [comment]
I think tracking the generated man is the way to go. It is still human readable even though it is generated, and so can serve as the source code of the man page when building.

**Comment by vivien** [comment]
I plan to switch all the documentation to asciidoctor. But in the meantime, tracking the generated doc seems totally legit to me if it makes packages lighter.
